doc/guides/dts/index.rst

```diff
@@ -346,44 +346,635 @@ Below is sample header content generated for this node, in
+
+       i2c0: i2c@40066000 {
+               compatible = "nxp,kinetis-i2c";
+               status = "okay";
+               #address-cells = <1>;
+               #size-cells = <0>;
+               reg = <0x40066000 0x1000>;
+
+               fxos8700@1d {
+                       compatible = "nxp,fxos8700";
+                       reg = <0x1d>;
+               };
+       };
+
+    For the ``i2c@40066000`` node above, the compatible string is
+    ``"nxp,kinetis-i2c"``. This gets converted to ``NXP_KINETIS_I2C`` by
+    uppercasing and replacing non-alphanumeric characters with underscores.
+    Adding the unit address gives the node identifier
+    ``DT_NXP_KINETIS_I2C_40066000``.
+
+    For the ``fxos8700@1d`` node above, since the binding specifies ``on-bus: i2c``
+    the bus portion of the node identifier will be: ``DT_NXP_KINETIS_I2C_40066000``.
+    The device node portion of the identifier will be ``NXP_FXOS8700_1D``.  The full
+    node identifier will be: ``DT_NXP_KINETIS_I2C_40066000_NXP_FXOS8700_1D``.
+
+    .. code-block:: none
+
+       ethernet@400c0004 {
+               compatible = "nxp,kinetis-ethernet";
+               reg = <0x400c0004 0x620>;
+               status = "okay";
+               ptp {
+                       compatible = "nxp,kinetis-ptp";
+                       status = "okay";
+                       interrupts = <0x52 0x0>;
+               };
+       };
 
-The ``*_BASE_ADDRESS`` part of the identifier is a fixed identifier generated
-from the special ``reg`` property on the node. ``*_SIZE`` is also generated
-from ``reg``. Other suffixes, like ``*_LABEL``, are generated directly from the
-property name.
+    For the ``ptp`` node above, since the node has no unit address, the unit
+    address portion will combine the parent's unit address and the node's
+    name.  The unit address portion of the identifier will be: ``400C0004_PTP``.
+    The full node identifier that combines the node compatible and unit address
+    will be: ``DT_NXP_KINETIS_PTP_400C0004_PTP``.
+
+    .. code-block:: none
+
+       soc {
+              temp1 {
+                      compatible = "nxp,kinetis-temperature";
+                      status = "okay";
+              };
+       };
+
+    For the ``temp1`` node above, since the node has no unit address the unit
+    address portion and the parent has no unit address we will utilize the
+    nodes name ``TEMP1`` as the unit address portion.
+    The full node identifier that combines the node compatible and unit address
+    will be: ``DT_NXP_KINETIS_TEMPERATURE_TEMP1``.
+
+``DT_INST_<instance-no.>_<compatible>``
+    The compatible string for the node together with an instance number.
+
+    The instance number is a unique index among all enabled
+    (``status = "okay"``) nodes that have a particular compatible string,
+    counting from zero. For example, if there are two enabled nodes that have
+    ``compatible = "foo,uart"``, then these node identifiers get generated:
```

Comment:
So e.g. the `DT_INST_0_FOO_UART` below is a prefix, not an identifier. `INST_0_FOO_UART` is the identifier.